### ISSUE TITLE
Creating exercises

### ISSUE THREAD

**Issue by tobyhodges** [carpentries-incubator/cwl-novice-tutorial#7]
Now that we have [learner profiles](https://carpentries-incubator.github.io/cwl-novice-tutorial/audience/) and [learning objectives]() drafted, as well as some [concept maps](https://docs.google.com/presentation/d/1aVdK8LHkgtESBunCQ-p7XmEl8NB9XbgDsH67X0_2HWg/edit#slide=id.g72208cbc10_0_264) for the tutorial, it's time to start creating some exercises for formative assessment 🎉 

A good starting point would probably be to aim for (at least) __one exercise per sub-point__ in the list of learning objectives (currently on the [tutorial landing page](https://carpentries-incubator.github.io/cwl-novice-tutorial/index.html)). We don't necessarily need exercises for the "lesson-level" objectives - ideally these will be covered by those for the more fine-grained objectives - but, if you have a good idea for something like a "capstone" exercise for assessing these higher-level learning objectives, that's welcome too 🙂

## Background Reading

For some background and guidance about what kind of exercises we should be aiming for, I recommend the following resources (please post additional suggestions in the comments):

- The [Exercise Types](http://teachtogether.tech/en/index.html#s:exercises) chapter of Greg Wilson's [Teaching Tech Together](http://teachtogether.tech/) (TtT)
- See examples of good exercises in these lessons:
  - [Storing Multiple Values in Lists](https://swcarpentry.github.io/python-novice-inflammation/05-lists/index.html) from the Software Carpentry _Programming with Python_ lesson
  - [Navigating Files and Directories](https://datacarpentry.org/shell-genomics/02-the-filesystem/index.html) from the Data Carpentry _Introduction to Command Line for Genomics_ lesson
  - [Libraries](https://librarycarpentry.org/lc-python-intro/06-libraries/index.html) from the Library Carpentry lesson _Python Intro for Libraries_

If you're interested in the theory and motivation behind creating these kinds of exercises for _formative assessment_, I recommend reading the following:

- The [Mental Models and Formative Assessment](http://teachtogether.tech/#s:models) chapter of Teaching Tech Together
- And the [Building Skill with Practice](https://carpentries.github.io/instructor-training/02-practice-learning/index.html) episode of [The Carpentries Instructor Training Curriculum](https://carpentries.github.io/instructor-training/), which covers much the same material as the TtT chapter linked above, but is also a good example of the kind of lesson material we want to create.

Lastly: if you're interested to learn more about the process we're following to develop this tutorial, then the chapter [A Lesson Design Process](http://teachtogether.tech/#s:process) from TtT is a great reference.

## Practicalities

- because we're working from The Carpentries' lesson template, [this guide](https://github.com/dmgt/swc_github_flow/blob/master/for_novice_contributors.md#making-lesson-changes-with-github) for contributing to their lessons is a good reference for this project too
  - more detail in their [Curriculum Development Handbook](https://carpentries.github.io/curriculum-development/technological-introductions.html)
   - we'll also have a call (scheduled for 14 May 2020) to go through the infrastructure and the contribution process in detail
  - anyway, please don't worry about getting something wrong! there'll be time to polish the material later, and __we want you to contribute__ and are more than happy to help you with that
- please create one Pull Request per draft exercise, including mention of which learning objective the exercise would assess
- write draft exercises in the episode file containing the relevant objective
  - [update] placeholder episode files for each set of objectives can be found in the `_episodes` directory. Please add your exercises to these files. Markdown documents saved here will become episode pages, and be added to the "Episodes" dropdown menu in the rendered lesson.

**Comment by swzCuroverse** [comment]
A note that we will be working on adding exercises and working to finish this tutorial as part of our work for the CZI EOSS grant.  More to come...

**Comment by tobyhodges** [comment]
This sounds very exciting, @swzCuroverse

**Comment by swzCuroverse** [comment]
Yes! @tobyhodges , @mr-c is helping as well as @tetron. I would love for you to review our work as we move along if you have bandwidth.  @tetron works at Curii with me (@mr-c works closely with us and helped us write the grant) so everyone is in the proverbial loop :)

**Comment by tobyhodges** [comment]
Very happy to review anything you come up with, time permitting. 

(BTW this lesson would fit nicely in The Carpentries Incubator while it's being developed, but I understand why you might prefer to keep it within the CWL ecosystem.)

**Comment by swzCuroverse** [comment]
I believe our plan is to keep it in the Carpentries style in the Carpentries system if at all possible.  So, let's see about that!

**Comment by tobyhodges** [comment]
:+1: Feel free to ping me if & when you need more information

**Comment by skanwal** [comment]
Hi @swzCuroverse 

I have been keen to find sometime to help with drafting a couple exercises. Initially, planning to start with https://github.com/common-workflow-lab/cwl-novice-tutorial/issues/8 and then possibly also look into https://github.com/common-workflow-lab/cwl-novice-tutorial/blob/gh-pages/_episodes/outputs.md.

Please let me know if this is fine? Happy to take inputs at any point.

**Comment by swzCuroverse** [comment]
Hi @skanwal that sounds great.  We have had a bi-weekly meeting where we have been meeting about our work - perhaps you could join us and talk about what you wanted to work on and we can sync up?  I can PM you?  We can reschedule the time to work with your time zone in Australia.

**Comment by swzCuroverse** [comment]
In the meantime, feel free to work and we can offer comments via github, etc :)

**Comment by skanwal** [comment]
@swzCuroverse - sounds great, I'll be keen to join bi-weekly discussions as well and thanks for considering time-zone difference. 
My email ID is kanwals@unimelb.edu.au. Please feel free to send me an email and we can hopefully workout a time that suits everyone. 

I am going through recent PRs from @bcosc and feedback from Michael and Toby. Will start working on a draft exercise in coming week.

**Comment by mr-c** [comment]
FYI, there will be a hackathon for this effort on March 22 and 23rd. Registration will open next week and in the mean time we are gathering details over at https://github.com/common-workflow-lab/cwl-novice-tutorial/wiki/2021-Lesson-Hackathon

**Comment by skanwal** [comment]
@mr-c thanks for this info.
I have marked these dates in my calendar. The sessions are starting an hour before midnight in Australia (Melbourne) but will try best to attend as I have grasped the main understanding, format and the concept behind the tutorial exercises  through [script episode](https://github.com/common-workflow-lab/cwl-novice-tutorial/pull/20).

Also, noticed a small typo in the date mentioned on the page for day2. It was `2021-02-23`, instead of `2021-03-23`. I was able to directly edit the page, hence updated it (hoping there are no issues?)
